tests/routers/test_statistics.py

```diff
@@ -0,0 +1,434 @@
+    assert response.status_code == 200
+    assert response.json() == {}
+
+    response = client.get("/api/v1/statistics/record/score-set/publication-identifiers")
+    assert response.status_code == 200
+    assert response.json() == {}
+
+    response = client.get("/api/v1/statistics/record/score-set/keywords")
+    assert response.status_code == 200
+    assert response.json() == {}
+
+    response = client.get("/api/v1/statistics/record/score-set/doi-identifiers")
+    assert response.status_code == 200
+    assert response.json() == {}
+
+    response = client.get("/api/v1/statistics/record/score-set/raw-read-identifiers")
+    assert response.status_code == 200
+    assert response.json() == {}
+
+
+## Test target accession statistics
+@pytest.mark.parametrize(
+    "field_value",
+    TARGET_ACCESSION_FIELDS,
+)
+def test_target_accession_statistics(client, setup_router_db, field_value, data_files):
+    """Test target accession statistics endpoint for published score sets."""
+    experiment = create_experiment(client)
+    with patch.object(cdot.hgvs.dataproviders.RESTDataProvider, "_get_transcript", return_value=TEST_CDOT_TRANSCRIPT):
+        score_set = create_acc_score_set_with_variants(client, experiment["urn"], data_files / "scores_acc.csv")
+
+    camelized_field_value = camelize(field_value.replace("-", "_"))
+    publish_score_set(client, score_set["urn"])
+    response = client.get(f"/api/v1/statistics/target/accession/{field_value}")
+
+    assert response.status_code == 200
+    assert TEST_MINIMAL_ACC_SCORESET["targetGenes"][0]["targetAccession"][camelized_field_value] in response.json()
+    assert response.json()[TEST_MINIMAL_ACC_SCORESET["targetGenes"][0]["targetAccession"][camelized_field_value]] == 1
+
+
+def test_target_accession_invalid_field(client):
+    """Test target accession statistic response for an invalid target accession field."""
+    response = client.get("/api/v1/statistics/target/accession/invalid-field")
+    assert response.status_code == 422
+    assert response.json()["detail"][0]["loc"] == ["path", "field"]
+    assert response.json()["detail"][0]["ctx"]["enum_values"] == TARGET_ACCESSION_FIELDS
+
+
+def test_target_accession_empty_field(client):
+    """Test target accession statistic response for an empty field."""
+    response = client.get("/api/v1/statistics/target/accession/")
+    assert response.status_code == 404
+    assert response.json()["detail"] == "Not Found"
+
+
+## Test target sequence statistics
+@pytest.mark.parametrize(
+    "field_value",
+    TARGET_SEQUENCE_FIELDS,
+)
+def test_target_sequence_statistics(client, setup_router_db, field_value, data_files):
+    """Test target sequence statistics endpoint for published score sets."""
+    experiment = create_experiment(client)
+    score_set = create_seq_score_set_with_variants(client, experiment["urn"], data_files / "scores.csv")
+
+    camelized_field_value = camelize(field_value.replace("-", "_"))
```

Comment:
to be specific, I don't think it would be a terrible idea to rip out the last 6 lines of each of these and just swap arround sequence/accession targetSequence/targetAccession. up to you

Reply:
Makes sense, I ended up pulling out a lot of the experiment / score set setup into their own fixtures and then just making a helper function to do the assert statements. They are all pretty much the same so that eliminated a lot.